packages/babel-plugin-proposal-optional-chaining/src/index.js

```diff
@@ -23,14 +23,23 @@ export default declare((api, options) => {
 
     visitor: {
       "OptionalCallExpression|OptionalMemberExpression"(path) {
-        const { parentPath, scope } = path;
+        const { scope } = path;
+        let parentPath;
+        // unwrap parenthesis around parent path
+        for (
+          { parentPath } = path;
+          parentPath.type === "ParenthesizedExpression";
+          { parentPath } = parentPath
+        );
         let isDeleteOperation = false;
         const optionals = [];
 
         let optionalPath = path;
         while (
           optionalPath.isOptionalMemberExpression() ||
-          optionalPath.isOptionalCallExpression()
+          optionalPath.isOptionalCallExpression() ||
+          optionalPath.isParenthesizedExpression() ||
+          optionalPath.isTSNonNullExpression()
```

Comment:
By doing so we also fix cases like `a!!?.b` are not transformed currently.

Reply:
@oliverdunk Should we also check for this in your PR?

Reply:
Makes sense, I'll add that now 👍